src/vs/base/common/lifecycle.ts

```diff
@@ -430,6 +430,34 @@ export class DisposableStore implements IDisposable {
 
 		return o;
 	}
+
+	/**
+	 * Deletes a disposable from store and disposes of it. This will not throw or warn and proceed to dispose the
+	 * disposable even when the disposable is not part in the store.
+	 */
+	public delete<T extends IDisposable>(o: T): void {
+		if (!o) {
+			return;
+		}
+		if ((o as unknown as DisposableStore) === this) {
+			throw new Error('Cannot dispose a disposable on itself!');
```

Comment:
Does this happen in practice? If so, I think the error message should be adjusted and `BugIndicatingError` should be used.

Reply:
🤷 just doing what `add` does

Reply:
At least the error message is weird.